apps/api/src/app/environments-v1/environments-v1.controller.ts

```diff
@@ -102,7 +97,7 @@ export class EnvironmentsControllerV1 {
         organizationId: user.organizationId,
         color: body.color,
         system: false,
-        returnApiKeys: isRbacEnabled ? user.permissions.includes(PermissionsEnum.API_KEY_READ) : true,
+        returnApiKeys: canAccessApiKeys,
```

Comment:
Initially I've tried to do it in reverse - always omitting secrets unless explicitly set to retrieve on repository level, but it turned out to be a huge refactoring, so I ended up doing this with a task in backlog:

https://linear.app/novu/issue/NV-5968/create-v2-versions-of-environments-and-integration-conrollers